### ISSUE TITLE
Report on installation with an error, not seems to be a blocking one

### ISSUE THREAD

**Issue by jhenligne** [ElvishJerricco/reflex-project-skeleton#13]
Hi there,
Thanks for this project, I expect it to help me spend less time in settings so as to focus on code.

Install
---
I installed reflex-project-skeleton this way:
```
$ git clone --recurse-submodules git@github.com:ElvishJerricco/reflex-project-skeleton.git
$ ./reflex-platform/try-reflex
```
And got this **error**:
`error: getting status of '/Users/kermit/Desktop/Dev/Haskell/reflex-project-skeleton/reflex-platform/gc-roots/shell.drv': No such file or directory`

Looking at `gc-root` directory:
```
$ ls -lah reflex-platform/gc-roots/
total 16
drwxr-xr-x   4 kermit  staff   136B 21 mar 09:17 .
drwxr-xr-x  78 kermit  staff   2,6K 21 mar 09:18 ..
-rw-r--r--   1 kermit  staff    25B 21 mar 09:17 .gitignore
-rw-r--r--   1 kermit  staff   426B 21 mar 09:17 README.md
```
It may be worth mentionning reflex-platform was already installed before.

Building
---
Going further anyway to see if it is a blocking error:
```
$ ./cabal new-build all
...
Linking /Users/kermit/Desktop/Dev/Haskell/reflex-project-skeleton/dist-newstyle/build/x86_64-osx/ghc-8.0.2/frontend-0.1.0.0/c/frontend/build/frontend/frontend ...
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_NSURL", referenced from:
      objc-class-ref in libHSjsaddle-wkwebview-0.9.4.0-2SHufpCZa9lIA8lareALCO.a(WKWebView.o)
ld: symbol(s) not found for architecture x86_64
clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
`cc' failed in phase `Linker'. (Exit code: 1)
cabal: Failed to build exe:frontend from frontend-0.1.0.0.
```
I know this one so I added -dynamic flag to `frontend.cabal`:
```
executable frontend
  main-is:             Main.hs
  ghc-options:         -Wall -O2 -dynamic -threaded -rtsopts -with-rtsopts=-N
```
Which solves this problem.

`$ ./cabal-ghcjs new-build all` runs fine too, `$ nix-build` too.

Testing results
---
```
$ backend-result/bin/backend
Hello, Haskell!
```
`$ open frontend-result/bin/frontend.jsexe/index.html` opens a browser tab with text `hi`.
`$ native-frontend-result/bin/frontend` opens an empty window. Is it normal? I expected `hi` to be displayed too.

Conclusion
---
- This error is spurious in my case but I don't know if it would impact other configurations.
- Warning MacOS users may be useful for those installing reflex directly with `reflex-project-skeleton`
- Is the result for `$ nix-build -o native-frontend-result -A ghc.frontend` conform to expectation?

**Comment by ElvishJerricco** [comment]
I think most of this will be solved by updating reflex-platform. Can you try going into the reflex-platform directory, pulling, and checking out develop?

**Comment by jhenligne** [comment]
Thanks, I tried it but it changed nothing.

Later I recalled getting the same empty window while trying `reflex-todomvc` with `./try-reflex` (just after installing `reflex-platform`), so it may not be related to `reflex-project-skeleton`.

I raised an issue about it and will come back to you when I have an update ([First 2 examples at prompt of ./try-reflex are broken #270](https://github.com/reflex-frp/reflex-platform/issues/270)).

**Comment by ElvishJerricco** [comment]
@jhenligne I believe most of this issue is due to compatibility between reflex-platform and Nix 2.0 (which you appear to be using). Would you mind moving these issues over there?

**Comment by alexfmpe** [comment]
```error: getting status of '/Users/kermit/Desktop/Dev/Haskell/reflex-project-skeleton/reflex-platform/gc-roots/shell.drv': No such file or directory```
Sounds like https://github.com/reflex-frp/reflex-platform/issues/250.

**Comment by jhenligne** [comment]
Thanks @alexfmpe, [reflex-frp/reflex-platform#250](https://github.com/reflex-frp/reflex-platform/issues/250) and PR [Fix try-reflex under nix 1.12/2.0 #221](https://github.com/reflex-frp/reflex-platform/pull/221) corrected the `shell.drv` error.
@ElvishJerricco I still have my issues described in my previous [comment](https://github.com/ElvishJerricco/reflex-project-skeleton/issues/13#issuecomment-375197630) but as it is not related to `reflex-project-skeleton` I close this issue, thanks for your help.
